roles/openshift_registry/tasks/main.yml

```diff
@@ -7,22 +7,10 @@
   command: >
     {{ openshift.common.admin_binary }} registry
     --create --service-account=registry {{ _oreg_selector }}
+    --replicas={{ openshift.master.infra_nodes | length }}
     --credentials={{ openshift_master_config_dir }}/openshift-registry.kubeconfig {{ _oreg_images }}
   register: _oreg_results
   changed_when: "'service exists' not in _oreg_results.stdout"
 
-- name: Determine if nfs volume is already attached
-  command: "{{ openshift.common.client_binary }} get -o template dc/docker-registry --template=\\{\\{.spec.template.spec.volumes\\}\\}"
-  register: registry_volumes_output
-  when: attach_registry_volume | bool
-
-- set_fact:
-    volume_already_attached: "{{ 'server:' + nfs_host in registry_volumes_output.stdout and 'path:' + registry_volume_path in registry_volumes_output.stdout }}"
-  when: attach_registry_volume | bool
-
-- name: Add nfs volume to dc/docker-registry
-  command: >
-    {{ openshift.common.client_binary }} volume dc/docker-registry
-    --add --overwrite --name=registry-storage --mount-path=/registry
-    --source='{"nfs": {"server": "{{ nfs_host }}", "path": "{{ registry_volume_path }}"}}'
-  when: attach_registry_volume | bool and not volume_already_attached | bool
+- include: nfs.yml
```

Comment:
Would it be better to have a openshift_pvs or openshift_persistent_volumes role that handles the importing/creation of the pvs and then just have the openshift_registry (and eventually metrics, logging, etc) just use a consistent name for the created PV?